src/SonarScanner.MSBuild.Common/TFS/BuildSettings.cs

```diff
@@ -296,8 +296,8 @@ public static BuildSettings CreateNonTeamBuildSettingsForTesting(string analysis
             {
                 BuildEnvironment = BuildEnvironment.NotTeamBuild,
                 AnalysisBaseDirectory = analysisBaseDirectory,
-                SonarScannerWorkingDirectory = Directory.GetParent(analysisBaseDirectory).FullName,
-                SourcesDirectory = Directory.GetParent(analysisBaseDirectory).FullName
+                SonarScannerWorkingDirectory = Directory.GetParent(analysisBaseDirectory)!.FullName,
+                SourcesDirectory = Directory.GetParent(analysisBaseDirectory)!.FullName
```

Comment:
What about this instead:
```suggestion
                SonarScannerWorkingDirectory = Directory.GetParent(analysisBaseDirectory)?.FullName ?? throw new ...,
                SourcesDirectory = Directory.GetParent(analysisBaseDirectory)!.FullName
```